pkg/network/tracer/tracer.go

```diff
@@ -313,6 +320,9 @@ func (t *Tracer) storeClosedConnections(connections []network.ConnectionStats) {
 		t.addProcessInfo(cs)
 
 		tracerTelemetry.closedConns.Inc(cs.Type.String())
+		if kprobe.FailedConnectionsSupported(t.config) {
```

Comment:
I think not having this check here would be cleaner. It is a little clunky to check this every time you call `GetFailedConnections`.

Reply:
I only added this because in `t.ebpfTracer.GetFailedConnections().MatchFailedConn(cs)`, `GetFailedConnections can return nil and thus NPE. It would be nice to have a Kotlin-esque notion of the `?` operator, so I could do 
``t.ebpfTracer.GetFailedConnections().?MatchFailedConn(cs)` which no-ops if the result is null instead of NPE

Reply:
You could return a no-op instance from `GetFailedConnections()`, which also means you would add an interface for `FailedConnections`, which I think would make this even cleaner.

Reply:
fixed with some null checks